src/npc.cpp

```diff
@@ -1198,13 +1203,22 @@ void npc::place_on_map()
     debugmsg( "Failed to place NPC in a valid location near (%d,%d,%d)", posx(), posy(), posz() );
 }
 
-skill_id npc::best_skill() const
+//selector = 0 is used for NPCs retreating and training skills
+//selector = 1 is used for NPC dialogue and simulating NPC combat
+//selector = 2 is used for determining which weapon an NPC spawns with
+skill_id npc::best_combat_skill( int selector ) const
 {
     int highest_level = std::numeric_limits<int>::min();
     skill_id highest_skill( skill_id::NULL_ID() );
 
     for( const auto &p : *_skills ) {
         if( p.first.obj().is_combat_skill() ) {
+            if( ( ( selector >= 1 ) && ( p.first == skill_dodge || p.first == skill_gun ||
+                                         p.first == skill_melee ) )
+                || ( ( selector >= 2 ) && ( p.first == skill_unarmed || p.first == skill_launcher ) ) ) {
```

Comment:
this means that NPC can't spawn with a launcher? Could you maybe unfold this in a switch; where for each value you do the loop and ignore a different set of skills? It would be more code but probably easier to read for anyone coming after you. Also it would be nice to use something more explicit than a int like enum in the function calls

Reply:
Thanks for the input, I reworked it a little and I think it looks cleaner now but let me know if it still needs changes.

Also yeah, an NPC can't spawn with a launcher but that's how it worked before I started messing around with things. That was also in the suggestion Erik gave in the original issue.